src/api/operations.ts

```diff
@@ -181,36 +183,90 @@ export const drawRectangle = (options: {
   rotate: Rotation;
   xSkew: Rotation;
   ySkew: Rotation;
+  borderRadius?: number | PDFNumber;
   borderLineCap?: LineCapStyle;
   borderDashArray?: (number | PDFNumber)[];
   borderDashPhase?: number | PDFNumber;
   graphicsState?: string | PDFName;
-}) =>
-  [
+}) => {
+  const {x: xN, y: yN, borderRadius: rN, width: wN, height: hN} = options;
+  const [x, y, r, w, h] = [
+    asNumber(xN), asNumber(yN), asNumber(rN || 0), asNumber(wN), asNumber(hN),
+  ];
+  const c = r * (1.0 - KAPPA);
+  return [
     pushGraphicsState(),
     options.graphicsState && setGraphicsState(options.graphicsState),
     options.color && setFillingColor(options.color),
     options.borderColor && setStrokingColor(options.borderColor),
     setLineWidth(options.borderWidth),
     options.borderLineCap && setLineCap(options.borderLineCap),
     setDashPattern(options.borderDashArray ?? [], options.borderDashPhase ?? 0),
-    translate(options.x, options.y),
+    translate(x, y),
     rotateRadians(toRadians(options.rotate)),
     skewRadians(toRadians(options.xSkew), toRadians(options.ySkew)),
-    moveTo(0, 0),
-    lineTo(0, options.height),
-    lineTo(options.width, options.height),
-    lineTo(options.width, 0),
+    moveTo(r, 0),
+    lineTo(w - r, 0),
+    r > 0 ? appendBezierCurve(w - c, 0, w, c, w, r) : null,
+    lineTo(w, h - r),
+    r > 0 ? appendBezierCurve(w, h - c, w - c, h, w - r, h) : null,
+    lineTo(r, h),
+    r > 0 ? appendBezierCurve(c, h, 0, h - c, 0, h - r) : null,
+    r > 0 ? lineTo(0, r) : null,
+    r > 0 ? appendBezierCurve(0, c, c, 0, r, 0) : null,
     closePath(),
-
     // prettier-ignore
     options.color && options.borderWidth ? fillAndStroke()
-  : options.color                      ? fill()
-  : options.borderColor                ? stroke()
-  : closePath(),
+    : options.color                      ? fill()
+    : options.borderColor                ? stroke()
+    : null,
 
     popGraphicsState(),
   ].filter(Boolean) as PDFOperator[];
+};
+
+export const drawLines = (options: {
```

Comment:
See my other comment for why I don't think that should be done.

Perhaps a counter-suggestion: add a drawPolygon which calls drawLines and defaults closePaths to true?

I added this function because I'm using it for several different things, including drawing checkmarks and partial outlines -- none of the cases I am using would "drawPolygon" be an appropriate name for what I am doing and in none of them would closing the path be desired.